File: example/dart/library/import/stack1.dart
library stack1;
final _contents = [];
get isEmpty => _contents.isEmpty;
get last => isEmpty ? throw 'Cannot get top of empty stack' : _contents.last;
get pop => isEmpty ? throw 'Cannot pop empty stack' : _contents.removeLast();
push(e) {
  _contents.add(e);
  return e;
}

File: example/typescript/library/import/stack1.ts
const _contents = []
const isEmpty = () => _contents.length === 0
const last = () => isEmpty() ? new Error('Cannot get top of empty stack') : _contents.reverse()[0]
const pop = () => isEmpty() ? new Error('Cannot pop empty stack') : _contents.pop()
const push = (e) => {
  _contents.push(e)
  return e
}

export {
  last,
  pop,
  push,
}